pkg/apis/provisioning/v1alpha5/provisioner.go

```diff
@@ -43,6 +43,35 @@ type ProvisionerSpec struct {
 	TTLSecondsUntilExpired *int64 `json:"ttlSecondsUntilExpired,omitempty"`
 	// Limits define a set of bounds for provisioning capacity.
 	Limits *Limits `json:"limits,omitempty"`
+	// InstanceTypeFilter allows filtering the instance types provided by the cloud provider.
+	InstanceTypeFilter *InstanceTypeFilter `json:"instanceTypeFilter,omitempty"`
+}
+
+// InstanceTypeFilter is the schema for the instance type filtering
+type InstanceTypeFilter struct {
+	// CPUCount allows filtering instance types by the min and maximum number of CPUs on the instance type with the
+	// minimum and maximum values being inclusive.
+	CPUCount *MinMax `json:"cpuCount,omitempty"`
+	// MemoryMiB allows filtering instance types by the min and maximum MiB of memory on the instance type with the
+	// minimum and maximum values being inclusive.
+	MemoryMiB *MinMax `json:"memoryMiB,omitempty"`
+	// MemoryMiBPerCPU allows filtering instance types by the min and maximum MiB of memory per on the instance type
+	// with the minimum and maximum values being inclusive.
+	MemoryMiBPerCPU *MinMax `json:"memoryMiBPerCPU,omitempty"`
+	// NameMatchExpressions are regular expressions to match against instance types names.  If no expressions are
```

Comment:
I like this! I was about to ask how we'd handle special CP specific instance type selection, but giving this at the top-level basically takes care of it. nice!